libraries/chain/apply_context.cpp

```diff
@@ -289,7 +289,7 @@ void apply_context::require_recipient( account_name recipient ) {
          schedule_action( action_ordinal, recipient, false )
       );
 
-      if (auto dm_logger = control.get_deep_mind_logger()) {
+      if (auto dm_logger = control.get_deep_mind_logger(); dm_logger && !trx_context.is_read_only()) {
```

Comment:
Thanks for your suggestion. Since read-only is an exception to deep-mind logging, it helps to understand the code by specifically showing when we don't do deep-mind logging.